lib/mongo_oplog_backup/rotate.rb

```diff
@@ -0,0 +1,89 @@
+require 'pathname'
+
+module MongoOplogBackup
+  class Rotate
+    attr_reader :config, :backup_list
+    DAY = 86400
+    RECOVERY_POINT_OBJECTIVE = 32 * DAY # Longest month + 1
+    KEEP_MINIMUM_SETS = 2 # Current & Previous
+
+    def initialize(config)
+      @config = config
+      @dry_run = !!@config.options[:dryRun]
+      @backup_list = Pathname.new(@config.backup_dir).children.select(&:directory?)
+      if @config.options[:keepDays].nil?
+        @recovery_point_objective = RECOVERY_POINT_OBJECTIVE
+      else
+        @recovery_point_objective = @config.options[:keepDays] * DAY
+      end
+
+    end
+
+    def current_backup_name
+      if @current_backup_name.nil?
+        state_file = config.global_state_file
+        state = JSON.parse(File.read(state_file)) rescue nil
+        state ||= {}
+        @current_backup_name = state['backup']
+      end
+      @current_backup_name
+    end
+
+    def perform
+      lock(config.global_lock_file) do
```

Comment:
Is the idea that this shouldn't run concurrently with backups?

Reply:
Correct, specifically not concurrent with Full backups to ensure that the state is stable. Perhaps overly cautious?